WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -205,7 +205,7 @@ private ConstructionState AddBlameScript(Round round, ConstructionState coinjoin
 			var diffMoney = Money.Satoshis(diff) - coinjoin.Parameters.FeeRate.GetFee(Config.BlameScript.EstimateOutputVsize());
 			if (diffMoney > coinjoin.Parameters.AllowedOutputAmounts.Min)
 			{
-				coinjoin = round.AddOutput(new TxOut(diffMoney, Config.BlameScript));
+				coinjoin = coinjoin.AddOutput(new TxOut(diffMoney, Config.BlameScript));
```

Comment:
Was the original an actual bug or just aesthetic? If it was a bug, may I ask what's the difference?

Reply:
The difference is while the state is immutable the round is not. Right now both are equivalent because everything is wrapped around a `lock` that guarantees that nobody had modified the round. Anyway, even if we never improve this code to make it more concurrent the coinjoin state is not the round but the CoinJoinState.